types/vote.go

```diff
@@ -61,6 +61,53 @@ type Vote struct {
 	ExtensionSignature []byte                `json:"extension_signature"`
 }
 
+func voteFromProto(pv *tmproto.Vote) (*Vote, error) {
+	if pv == nil {
```

Comment:
As a general rule, please don't inject nil checks for these. The pointer is the actionable type for a protobuf message, so if the caller passes nil here it's not a recoverable error and should probably be allowed to panic.

(A program that calls this function with a nil Vote message is already statically incorrect. If there needs to e a check, it should be on the caller's side. But the way we use it, that isn't necessary).

Reply:
I just kept the existing code as-is as far as possible, but I agree - I'll change this.